cmd/agent/app/reporter/grpc/builder_test.go

```diff
@@ -180,6 +189,39 @@ func TestProxyBuilder(t *testing.T) {
 			grpcBuilder: &ConnBuilder{CollectorHostPorts: []string{"localhost:0000"}, TLS: true, TLSCA: "/not/valid"},
 			expectError: true,
 		},
+		{
+			name: "with secure grpc connection and valid TLS Client settings",
+			grpcBuilder: &ConnBuilder{
+				CollectorHostPorts: []string{"localhost:0000"},
+				TLS:                true,
+				TLSCA:              tmpfile.Name(),
+				TLSCert:            "testdata/client.jaeger.io-client.pem",
```

Comment:
if we're going with separate  test files, it would be good to remove the temp file logic in L153-L165 and move `certPEM` to a file as well. Looks like it simplifies the test.